jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -247,9 +247,9 @@ private void analyzeFile(String pkgName, String inPath, boolean includeNonPublic
               DefinitelyDerefedParams analysisDriver = null;
               String sign = "";
               // Parameter analysis
-              if (mtd.getNumberOfParameters() > (mtd.isStatic() ? 0 : 1)) {
-                // Skip methods by looking at bytecode
-                try {
+              try {
+                if (mtd.getNumberOfParameters() > (mtd.isStatic() ? 0 : 1)) {
+                  // Skip methods by looking at bytecode
                   if (!CodeScanner.getFieldsRead(mtd).isEmpty()
                       || !CodeScanner.getFieldsWritten(mtd).isEmpty()
                       || !CodeScanner.getCallSites(mtd).isEmpty()) {
```

Comment:
In retrospect, not entirely sure what we are skipping here. It doesn't look like this skip logic applies to computing a method's _return_ nullability, just that of its arguments, _and_, if the logic here is to avoid methods that aren't called from somewhere else or assigned to a field, then we probably still shouldn't be skipping them in terms of analyzing libraries...

Reply:
I think the logic here is as follows: if a method inside its body does not read any fields, write any fields, or call any other methods, then it does not perform any explicit pointer dereferences.  Hence we will not be able to infer any `@NonNull` parameters for the method.  You would not want to perform the same optimization for return nullability, since then a method whose body is `return null` could be skipped.  We should probably add a comment to that effect here, assuming it makes sense to you, @lazaroclapp.

Reply:
Yup. Makes perfect sense. Will make sure to add a comment, though, as it isn't super intuitive at a glance.

Reply:
We could also extract the logic inside the conditional into a new method `containsNoDereferences(IMethod)` which might improve readability; your call

Reply:
Sure. Makes sense to me!